File: packages/jexlang-java/src/main/java/io/github/vinaykumar0339/functions/FuncRegistry.java
package io.github.vinaykumar0339.functions;

import io.github.vinaykumar0339.context.EvaluatorContext;
import io.github.vinaykumar0339.types.JexValue;

public interface FuncRegistry {
    boolean has(String name);
    JexValue call(String name, EvaluatorContext ctx, JexValue... args);
}

File: packages/jexlang-swift/jexlang-swift/eval/functions/FuncRegistry.swift
//
//  FuncRegistry.swift
//  jexlang-swift
//
//  Created by Vinay Kumar on 04/11/25.
//

import Foundation

public protocol FuncRegistry {
    func has(_ name: String) -> Bool
    func call(_ name: String, _ args: [JexValue]) -> JexValue
}
